File: backend/schemas/word.py
from pydantic import BaseModel


class PhraseItem(BaseModel):
    p_cn: str
    p_content: str


class HwdItem(BaseModel):
    hwd: str | None = None
    tran: str | None = None
    word: str | None = None


class RelWordGroup(BaseModel):
    Hwds: list[HwdItem]
    Pos: str


class SentenceItem(BaseModel):
    s_cn: str
    s_content: str


class SynonymGroup(BaseModel):
    Hwds: list[HwdItem]
    pos: str
    tran: str


class TranslationItem(BaseModel):
    pos: str
    tran_cn: str


class WordData(BaseModel):
    bookId: str | None = None
    phrases: list[PhraseItem] = []
    relWords: list[RelWordGroup] = []
    sentences: list[SentenceItem] = []
    synonyms: list[SynonymGroup] = []
    translations: list[TranslationItem] = []
    ukphone: str | None = None
    ukspeech: str | None = None
    usphone: str | None = None
    usspeech: str | None = None
    word: str


class WordLookupResponse(BaseModel):
    code: int = 200
    data: WordData
    message: str = "success"

File: src/types/word.ts
export interface PhraseItem {
  p_cn: string;
  p_content: string;
}

export interface HwdItem {
  hwd?: string;
  tran?: string;
  word?: string;
}

export interface RelWordGroup {
  Hwds: HwdItem[];
  Pos: string;
}

export interface SentenceItem {
  s_cn: string;
  s_content: string;
}

export interface SynonymGroup {
  Hwds: HwdItem[];
  pos: string;
  tran: string;
}

export interface TranslationItem {
  pos: string;
  tran_cn: string;
}

export interface WordLookupData {
  bookId: string | null;
  phrases: PhraseItem[];
  relWords: RelWordGroup[];
  sentences: SentenceItem[];
  synonyms: SynonymGroup[];
  translations: TranslationItem[];
  ukphone: string | null;
  ukspeech: string | null;
  usphone: string | null;
  usspeech: string | null;
  word: string;
}

export interface VocabItem {
  id: string;
  word: string;
  phonetic: string | null;
  pos: string | null;
  mean: string | null;
  trans: string | null;
  added: string | null;
  example: string | null;
  exampleTrans: string | null;
  mastery: number;
  masteryScore: number;
  lastReviewedAt: string | null;
  reviewCount: number;
}

export interface MasteryResult {
  mastery: number;
  masteryScore: number;
  reviewCount: number;
}
